### DISCUSSION TITLE
Internal training and knowledge sharing

### DISCUSSION THREAD

**Discussion by raven** [MobileNativeFoundation/discussions#95] [General]
I'm curious to hear how other organisations run internal training / knowledge sharing / up-or-cross skilling.

One of the problems we have been trying to solve is that we identified a skills gap of novice or beginner native mobile developers. We found that outside of our native mobile developers, there was little to no exposure to native mobile development, where as other disciplines like web front-end were able to contribute to backend or systems engineering tasks. 

- Do you have regular sessions that are applicable to all native mobile developers?
- Do you have discipline specific (iOS vs Android vs Web) sessions?
- Do you bring in external facilitators or run it in-house? 
- How is active engagement or attendance encouraged?
- Do you run cross-skilling training? e.g Internal courses or coaching for non native mobile developers to learn native mobile development?
- How T-shaped / cross-skilled are your native developers encouraged to be?

---

At REA for native mobile development we have Swift Dojos and Kotlin Dojos every fortnight where discipline specific topics are showcased. On the alternating fortnights we have Mobile Guild which hosts broader topics like what our different apps are tackling or experimenting with. We also have guilds for other disciplines that are not platform specific such as Functional Programming Guild. Beyond the guilds and dojos we also run Mobile Bootcamp (separate ones for each of iOS and Android), which is a multi-day course we run in-house that is designed for non native mobile developers to get an introduction to the languages, IDEs, toolchains and release processes. The bootcamps have gone a long way to addressing the 'zero understanding of native mobile development' for other developers within the organisation, and we have had a few people subsequently change roles to become native mobile developers also.

**Comment by shivankD** [comment]
Hi @raven ! We've got something similar here at ASB. We run weekly catchup sessions for Android and iOS devs (separate events) and then we have a chapter wide meeting every fortnight. The agendas are usually open for anyone and they're free to talk about whatever mobile related topic they like - proposals for improvement, future roadmap, experiences with a certain technology etc. 

We've got something in our pipeline to start an in-house bootcamp for non-native developers. Would love to pick your brains on that one and learn from your experiences!
